src/inmanta/protocol/openapi/converter.py

```diff
@@ -410,10 +412,49 @@ def _build_responses(self, url_method_properties: MethodProperties) -> Dict[str,
 
     def _build_return_value_wrapper(self, url_method_properties: MethodProperties) -> Optional[Dict[str, MediaType]]:
         return_type = inspect.signature(url_method_properties.function).return_annotation
-        if return_type is not None and return_type != inspect.Signature.empty:
-            return_properties: Optional[Dict[str, Schema]] = None
+
+        if return_type is None or return_type == inspect.Signature.empty:
+            return None
+
+        return_properties: Optional[Dict[str, Schema]] = None
+
+        if is_generic_type(return_type) and get_origin(return_type) == ReturnValue:
```

Comment:
In Python 3.9 (and possibly 3.7+) this condition will not match for the `ReturnValue` type (in contrast to e.g. `ReturnValue[int]`) because `get_origin` returns `None` if there are no args. 
```suggestion
        if return_type == ReturnValue or is_generic_type(return_type) and get_origin(return_type) == ReturnValue:
```